blobstore/blobstore.py

```diff
@@ -0,0 +1,42 @@
+from __future__ import absolute_import, division, print_function, unicode_literals
+
+class BlobStore(object):
+    """Abstract base class for all blob stores."""
+    def __init__(self):
+        pass
+
+    def list(self, prefix=None):
+        """Returns a list of all blob entries in a container that match a given prefix."""
+        raise NotImplementedError()
+
+    def get_presigned_url(self, objname):
+        """Retrieves a presigned url for blob objname.  Raises BlobNotFoundError if the blob is not present."""
+        raise NotImplementedError()
+
+    def set(self, objname, src_file_handle):
```

Comment:
This is for unit testing only, right? We wouldn't want to slurp blob data through the upload route in the actual API - we want some sort of smart client behavior based on either a redirect or temporary upload credentials returned by the upload route.